### PR TITLE
adding examples for tags in create_experiment and start_run

### LINKED ISSUE #6101: [FR] [Roadmap] Add examples of applying experiment and run tags during instantiation
## MLflow Roadmap Item

This is an MLflow Roadmap item that has been prioritized by the MLflow maintainers. We’ve identified this feature as a highly requested addition to the MLflow package based on community feedback. 
We're seeking a community contribution for the implementation of this feature and will enthusiastically support the development and review of a submitted PR for this.

## Contribution Note

As with other roadmap items, there may be a desire for multiple contributors to work on an issue. While we don’t discourage collaboration, we strongly encourage that a primary contributor is assigned to roadmap issues to simplify the merging process. 
The items on the roadmap are of a high priority. Due to the wide-spread demand of roadmap features, we encourage potential contributors to only agree to take on the work of creating a PR, making changes, and ensuring that test coverage is adequately created for the feature if they are willing and able to see the implementation through to a merged state. 

## Feature scope

This roadmap feature’s complexity is classified as:

- [X] `good-first-issue`: This feature is limited in complexity and effort required to implement.
- [ ] `simple`: This feature does not require a large amount of effort to implement and / or is clear enough to not need a design discussion with maintainers.
- [ ] `involved`: This feature will require a substantial amount of development effort but does not require an agreed-upon design from the maintainers. The feedback given during the PR phase may be involved and necessitate multiple iterations before approval. (Please bear with us as we collaborate with you to make a great contribution)
- [ ] `design-recommended`: This is a substantial feature that should have a design document approved prior to working on an implementation (to save your time, not ours). After agreeing to work on this feature, a maintainer will be assigned to support you throughout the development process.

## Proposal Summary

Add examples to and more throughly explain the options available during experiment creation and run context instantiation [here](https://mlflow.org/docs/latest/tracking.html#logging-functions). Explanation of available arguments with a short example of how to apply tags, names, and how to nest runs within this section are the limit of the scope of this FR.
An example of how to fill out the description section with reserved tag keys will be welcomed.

   Improve the visibility of these commonly misunderstood features of tag usage, description setting with reserved tag keys, and how to nest runs will benefit users.

> #### Why is this use case valuable to support for MLflow users in general? 
   The only visible part of MLflow to know that these features exist is the source code.

### What component(s), interfaces, languages, and integrations does this feature affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interfaces
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Languages 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by subramaniam02**
## Related Issues/PRs

closes #6101 

## What changes are proposed in this pull request?
expanded examples for `create_experiment` and `start_run` to have tags, descriptions

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [x] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by subramaniam02** [comment]
@harupy Can you review this PR ? Let me know if you need more comprehensive changes. Thanks !!

File: mlflow/tracking/fluent.py [hunk 1]
```diff
@@ -1166,7 +1177,11 @@ def create_experiment(
         import mlflow
 
         # Create an experiment name, which must be unique and case sensitive
-        experiment_id = mlflow.create_experiment("Social NLP Experiments")
+        experiment_id = mlflow.create_experiment(
+            "Social NLP Experiments",
+            artifact_location="file:///absolute-path/mlruns",
+            tags={"version": "v1", "priority": "P1"},
+        )
```
Comment (harupy) [inline mlflow/tracking/fluent.py:1184]: Can you update the output block below?

File: mlflow/tracking/client.py [hunk 2]
```diff
@@ -586,7 +586,11 @@ def create_experiment(
 
             # Create an experiment with a name that is unique and case sensitive.
             client = MlflowClient()
-            experiment_id = client.create_experiment("Social NLP Experiments")
+            experiment_id = client.create_experiment(
+                "Social NLP Experiments",
+                artifact_location="file:///absolute-path/mlruns",
+                tags={"version": "v1", "priority": "P1"},
+            )
```
Comment (harupy) [inline mlflow/tracking/client.py:593]: same here https://github.com/mlflow/mlflow/pull/6236/files#r923482608

File: mlflow/tracking/client.py [hunk 3]
```diff
@@ -586,7 +586,11 @@ def create_experiment(
 
             # Create an experiment with a name that is unique and case sensitive.
             client = MlflowClient()
-            experiment_id = client.create_experiment("Social NLP Experiments")
+            experiment_id = client.create_experiment(
+                "Social NLP Experiments",
+                artifact_location="file:///absolute-path/mlruns",
```
Comment (harupy) [inline mlflow/tracking/client.py:591]:
```suggestion
                artifact_location=Path.cwd().joinpath("mlruns"),
```

Comment (harupy) [reply mlflow/tracking/client.py:591]: Can you also add `from pathlib import Path`?

File: mlflow/tracking/fluent.py (hunk 1 above)
Comment (subramaniam02) [reply mlflow/tracking/fluent.py:1184]: my bad :( Will update.

File: mlflow/tracking/fluent.py [hunk 4]
```diff
@@ -218,14 +218,33 @@ def start_run(
         import mlflow
 
         # Create nested runs
-        with mlflow.start_run(run_name='PARENT_RUN') as parent_run:
+        experiment_id = mlflow.create_experiment("experiment1")
+        with mlflow.start_run(
+            run_name="PARENT_RUN",
+            experiment_id=experiment_id,
+            tags={"version": "v1", "priority": "P1"},
+            description="starting a parent for experiment {}".format(experiment_id),
```
Comment (harupy) [inline mlflow/tracking/fluent.py:226]:
```suggestion
            description="parent",
```

to simplify the example.

File: mlflow/tracking/fluent.py [hunk 5]
```diff
@@ -218,14 +218,33 @@ def start_run(
...
+        ) as parent_run:
             mlflow.log_param("parent", "yes")
-            with mlflow.start_run(run_name='CHILD_RUN', nested=True) as child_run:
+            with mlflow.start_run(
+                run_name="CHILD_RUN",
+                experiment_id=experiment_id,
+                description="child run of parent run: {}".format(parent_run.info.run_id),
```
Comment (harupy) [inline mlflow/tracking/fluent.py:232]:
```suggestion
                description="child",
```

File: mlflow/tracking/fluent.py [hunk 6]
```diff
@@ -1179,8 +1211,8 @@ def create_experiment(
 
         Name: Social NLP Experiments
         Experiment_id: 1
-        Artifact Location: file:///.../mlruns/1
-        Tags= {}
+        Artifact Location: file:///.../mlruns
+        Tags: {'version': 'v1', 'priority': 'P1', 'nlp.framework': 'Spark NLP'}
```
Comment (harupy) [inline mlflow/tracking/fluent.py:1215]: Where does `nlp.framework` come from?

File: mlflow/tracking/fluent.py [hunk 7]
```diff
@@ -218,14 +218,33 @@ def start_run(
...
                 mlflow.log_param("child", "yes")
 
-        print("parent run_id: {}".format(parent_run.info.run_id))
-        print("child run_id : {}".format(child_run.info.run_id))
+        print("parent run:")
+
+        print("run_id: {}".format(parent_run.info.run_id))
+        print("description: {}".format(parent_run.data.tags.get("mlflow.note.content")))
+        print("version tag value: {}".format(parent_run.data.tags.get("version")))
+        print("priority tag value: {}".format(parent_run.data.tags.get("priority")))
```
Comment (harupy) [inline mlflow/tracking/fluent.py:242]:
```suggestion
        print("description: {}".format(parent_run.data.tags))
```

Can we just print out the tags?

Comment (subramaniam02) [reply mlflow/tracking/fluent.py:242]: It prints a lot of system tags. I thought it might cause confusions to users.

Comment (harupy) [reply mlflow/tracking/fluent.py:242]: Make sense. Let's keep the current code.

File: mlflow/tracking/client.py [hunk 8]
```diff
@@ -581,12 +581,16 @@ def create_experiment(
 
         .. code-block:: python
             :caption: Example
-
+            from pathlib import Path
```
Comment (harupy) [inline mlflow/tracking/client.py:584]:
```suggestion

            from pathlib import Path
```

A new line needs to inserted.

File: mlflow/tracking/fluent.py [hunk 9]
```diff
@@ -235,11 +254,19 @@ def start_run(
...
+        parent run:
+        run_id: 8979459433a24a52ab3be87a229a9cdf
+        description: starting a parent for experiment 7
+        version tag value: v1
+        priority tag value: P1
+        --
+        child run:
+        run_id : 7d175204675e40328e46d9a6a5a7ee6a
+        --
+        Child runs of parent run_id: 8979459433a24a52ab3be87a229a9cdf
```
Comment (harupy) [inline mlflow/tracking/fluent.py:266]: Do we need this? The parent-run ID is printed in line 258.

File: mlflow/tracking/client.py (hunk 8 above)
Comment (subramaniam02) [reply mlflow/tracking/client.py:584]: I believe it does not cause any issue in rendering but more in the spirit of uniformity ?

Comment (harupy) [reply mlflow/tracking/client.py:585]: This breaks the documentation build:

https://app.circleci.com/pipelines/github/mlflow/mlflow/13128/workflows/ea7493a1-125c-4eb2-b305-edb9589c165b/jobs/23129

Comment (harupy) [reply mlflow/tracking/client.py:585]:
```
/home/circleci/project/mlflow/tracking/client.py:docstring of mlflow.tracking.client.MlflowClient.create_experiment:11: WARNING: Error in "code-block" directive:
invalid option block.

.. code-block:: python
    :caption: Example
    from pathlib import Path
    from mlflow import MlflowClient

    # Create an experiment with a name that is unique and case sensitive.
    client = MlflowClient()
    experiment_id = client.create_experiment(
        "Social NLP Experiments",
        artifact_location=Path.cwd().joinpath("mlruns").as_uri(),
        tags={"version": "v1", "priority": "P1"},
    )
    client.set_experiment_tag(experiment_id, "nlp.framework", "Spark NLP")

    # Fetch experiment metadata information
    experiment = client.get_experiment(experiment_id)
    print("Name: {}".format(experiment.name))
    print("Experiment_id: {}".format(experiment.experiment_id))
    print("Artifact Location: {}".format(experiment.artifact_location))
    print("Tags: {}".format(experiment.tags))
    print("Lifecycle_stage: {}".format(experiment.lifecycle_stage))

```

File: mlflow/tracking/fluent.py [hunk 10]
```diff
@@ -218,14 +218,33 @@ def start_run(
...
+
+        print("run_id: {}".format(parent_run.info.run_id))
+        print("description: {}".format(parent_run.data.tags.get("mlflow.note.content")))
+        print("version tag value: {}".format(parent_run.data.tags.get("version")))
+        print("priority tag value: {}".format(parent_run.data.tags.get("priority")))
+        print("--")
+        print("child run:")
+        print("run_id : {}".format(child_run.info.run_id))
         print("--")
+        print("child runs of parent run")
```
Comment (harupy) [inline mlflow/tracking/fluent.py:247]:
```python
        # Search all child runs with a parent id
        query = "tags.mlflow.parentRunId = '{}'".format(parent_run.info.run_id)
        results = mlflow.search_runs(filter_string=query)
        print("child runs")
        print(results[["run_id", "params.child", "tags.mlflow.runName"]])
```

File: mlflow/tracking/client.py (hunk 8 above)
Comment (subramaniam02) [reply mlflow/tracking/client.py:585]: I ran `make livehtml` and was able to see docs update. Nevertheless updated the code with newline. Will monitor docs ci job going forward. Thanks !

File: mlflow/tracking/fluent.py (hunk 9 above)
Comment (harupy) [reply mlflow/tracking/fluent.py:266]: `print("child runs of parent run")` doesn't print out `print("child runs of parent run")`. Can we fix?

File: mlflow/tracking/fluent.py [hunk 11]
```diff
@@ -218,28 +218,54 @@ def start_run(
...
 
-        parent run_id: 5ec0e7ae18f54c2694ffb48c2fccf25c
-        child run_id : 78b3b0d264b44cd29e8dc389749bb4be
+        parent run:
+        run_id: 8979459433a24a52ab3be87a229a9cdf
+        description: starting a parent for experiment 7
+        version tag value: v1
+        priority tag value: P1
+        --
+        child run:
```
Comment (harupy) [inline mlflow/tracking/fluent.py:263]:
```suggestion
        child runs:
```

**Comment by subramaniam02** [comment]
Seeing an issue in docs. WIll take a look deeper look on why example block is missing
Screen Shot 2022-07-18 at 10 47 05 PM

**Comment by harupy** [comment]
@subramaniam02 It renders fine here: https://output.circle-artifacts.com/output/job/eb310260-15f9-4b57-88ee-1d6b6f30ae58/artifacts/0/docs/build/html/python_api/mlflow.html

You can try running `make clean rsthtml`.

File: mlflow/tracking/fluent.py [hunk 12]
```diff
@@ -218,28 +218,54 @@ def start_run(
...
+        print("version tag value: {}".format(parent_run.data.tags.get("version")))
+        print("priority tag value: {}".format(parent_run.data.tags.get("priority")))
+        print("--")
+        print("child run:")
+        print("run_id : {}".format(child_run.info.run_id))
         print("--")
 
         # Search all child runs with a parent id
         query = "tags.mlflow.parentRunId = '{}'".format(parent_run.info.run_id)
         results = mlflow.search_runs(filter_string=query)
```
Comment (harupy) [inline mlflow/tracking/fluent.py:250]:
```suggestion
        results = mlflow.search_runs(experiment_ids=[experiment_id], filter_string=query)
```

**Review by harupy — APPROVED** [review]
LGTM!

File: mlflow/tracking/fluent.py [hunk 13]
```diff
@@ -218,28 +218,54 @@ def start_run(
...
-        print("child run_id : {}".format(child_run.info.run_id))
+        print("parent run:")
+
+        print("run_id: {}".format(parent_run.info.run_id))
+        print("description: {}".format(parent_run.data.tags.get("mlflow.note.content")))
+        print("version tag value: {}".format(parent_run.data.tags.get("version")))
+        print("priority tag value: {}".format(parent_run.data.tags.get("priority")))
+        print("--")
+        print("child run:")
+        print("run_id : {}".format(child_run.info.run_id))
```
Comment (harupy) [inline mlflow/tracking/fluent.py:245]:
```suggestion
```

Let's remove this.

Comment (subramaniam02) [reply mlflow/tracking/fluent.py:245]: I had same thought. But since I didnt hear any feedback, I left it out.

Comment (subramaniam02) [reply mlflow/tracking/fluent.py:245]: I recognize this PR has spawned another PR https://github.com/mlflow/mlflow/pull/6254

**Pull request merged**